crowdaction/main.go

```diff
@@ -0,0 +1,294 @@
+			action, err := json.Marshal(map[string]interface{}{"data": crowdaction})
+			if err != nil {
+				return events.APIGatewayProxyResponse{Body: err.Error(), StatusCode: http.StatusBadRequest}, nil
+			}
+			listCrowdaction = append(listCrowdaction, string(action))
+		}
+	}
+	body := "[" + strings.Join(listCrowdaction, ",") + "]"
+	return events.APIGatewayProxyResponse{
+		Body:       body,
+		StatusCode: http.StatusOK,
+	}, nil
+}
+
+//get details about a crowd action
+func getCrowdaction(crowdactionID string, req events.APIGatewayV2HTTPRequest) (events.APIGatewayProxyResponse, error) {
+
+	var crowdaction Crowdaction
+
+	val := prefixPKcrowdactionID + crowdactionID
+	out, err := getItem(val)
+
+	if err != nil {
+		body, err := json.Marshal(map[string]interface{}{"message": err.Error()})
+		if err != nil {
+			return events.APIGatewayProxyResponse{Body: err.Error(), StatusCode: http.StatusBadRequest}, nil
+		}
+		return events.APIGatewayProxyResponse{
+			Body:       string(body),
+			StatusCode: http.StatusInternalServerError,
+		}, nil
+	}
+
+	if out.Item == nil {
+		body, err := json.Marshal(map[string]string{"message": "crowdaction does not exist"})
+		if err != nil {
+			return events.APIGatewayProxyResponse{Body: err.Error(), StatusCode: http.StatusBadRequest}, nil
+		}
+		return events.APIGatewayProxyResponse{
+			Body:       string(body),
+			StatusCode: http.StatusNotFound,
+		}, nil
+	}
+
+	err = dynamodbattribute.UnmarshalMap(out.Item, &crowdaction)
+	if err != nil {
+		return events.APIGatewayProxyResponse{Body: err.Error(), StatusCode: http.StatusBadRequest}, nil
+	}
+
+	body, err := json.Marshal(map[string]interface{}{"data": crowdaction})
+	if err != nil {
+		return events.APIGatewayProxyResponse{Body: err.Error(), StatusCode: http.StatusBadRequest}, nil
+	}
+	return events.APIGatewayProxyResponse{
+		Body:       string(body),
+		StatusCode: http.StatusOK,
+	}, nil
+
+}
+
+func dynamodbClient() *dynamodb.DynamoDB {
+	sess := session.Must(session.NewSession())
+	return dynamodb.New(sess)
+}
+
+func getItem(val string) (*dynamodb.GetItemOutput, error) {
+
+	result, err := dbClient.GetItem(&dynamodb.GetItemInput{
+		TableName: &tableName,
+		Key: map[string]*dynamodb.AttributeValue{
+			"pk": {
+				S: aws.String(val),
+			},
+			"sk": {
+				S: aws.String(val),
+			},
+		},
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if result.Item == nil {
+		msg := "could not find crowdaction '" + val + "'"
+		return nil, errors.New(msg)
+	}
+
+	return result, nil
+}
+
+func getItems(pk, sk string) (*dynamodb.QueryOutput, error) {
+
+	var keyCond expression.KeyConditionBuilder
+
+	if sk == "" {
+		keyCond = expression.Key("pk").Equal(expression.Value(pk))
+	} else {
+		keyCond = expression.KeyAnd(
+			expression.Key("pk").Equal(expression.Value(pk)),
+			expression.Key("sk").LessThan(expression.Value(sk)),
+		)
+	}
+
+	expr, err := expression.NewBuilder().WithKeyCondition(keyCond).Build()
+	if err != nil {
+		return nil, err
+	}
+
+	/*
```

Comment:
Please remove code that has been commented out or annotate it with a TODO and explanation if it is needed later.